Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp

```diff
@@ -145,20 +173,57 @@ void ABenchmarkGymGameMode::Tick(float DeltaSeconds)
 
 		for (int i = AIControlledPlayers.Num() - 1; i >= 0; i--)
 		{
-			checkf(AIControlledPlayers[i].Controller, TEXT("Simplayer controller has been deleted."));
-			ACharacter* Character = AIControlledPlayers[i].Controller->GetCharacter();
+			AController* Controller = AIControlledPlayers[i].Key.Get();
+			int InfoIndex = AIControlledPlayers[i].Value;
+
+			checkf(Controller, TEXT("Simplayer controller has been deleted."));
+			ACharacter* Character = Controller->GetCharacter();
 			checkf(Character, TEXT("Simplayer character does not exist."));
-			int InfoIndex = AIControlledPlayers[i].Index;
 			UDeterministicBlackboardValues* Blackboard = Cast<UDeterministicBlackboardValues>(Character->FindComponentByClass(UDeterministicBlackboardValues::StaticClass()));
 			checkf(Blackboard, TEXT("Simplayer does not have a UDeterministicBlackboardValues component."));
-			
+
 			const FBlackboardValues& Points = PlayerRunPoints[InfoIndex % PlayerRunPoints.Num()];
 			Blackboard->ClientSetBlackboardAILocations(Points);
 		}
 		AIControlledPlayers.Empty();
 	}
+
+	ServerUpdateNFRTestMetrics();
 }
 
+void ABenchmarkGymGameMode::ServerUpdateNFRTestMetrics()
+{
+	float ClientRTTSeconds = 0.0f;
```

Comment:
Perhaps this is only valid if a number of clients are connected?
And I think we should be outputting the result of this at a regular interval.